lessons/Lesson-4-Ruby-Part-2.md

```diff
@@ -10,71 +11,93 @@ stringArray.forEach((str) => {
 })
 ```
 
-```
+```ruby
 # Next, in Ruby, with a block
 
 string_array = ["first", "second", "third"]
 
-string_array.each do | str | # you could also have { | str| puts str }
+string_array.each do |str|
   puts str
 end
 ```
-The appearance is a little different.  You surround the value that is to be passed on each iteration with vertical bars.  Some blocks are passed multiple values, as in the following:
+
+There are a couple of alternative syntaxes that do the same thing:
+
+```ruby
+string_array.each { |str| puts str }
+string_array.each { puts _1 } # `_1` refers to the first block argument
 ```
+
+The appearance is a little different.  You surround the value that is to be passed on each iteration with vertical bars.  Some blocks are passed multiple values, as in the following:
+
+```ruby
 this_hash = {first: 1, second: "two", third: :three }
 
-this_hash.each_pair do | key, value |
+this_hash.each_pair do |key, value|
   puts "the key is #{key}"
   puts "the value is #{value}"
 end
-
 ```
-You never use "return" in a block.  Sometimes a method may take a parameter as well as a block, as in the following:
 
-```
-my_array = [3, 4, 5]
-puts( my_array.reduce(7) do |total, element|
-  total += element
-end)
```

Comment:
Moved `reduce` further down because what the method does is a little complicated to start with if you don't have a functional background.